canvas/src/lib.rs

```diff
@@ -569,6 +569,29 @@ impl Path2D {
         }
     }
 
+    /// Add other path to this path.
+    ///
+    /// note: any non-closed subpaths will be closed
+    pub fn add_path(&mut self, mut other: Path2D) {
+        self.flush_current_contour();
+        other.flush_current_contour();
+        
+        // FIXME(s3bk) should consume segments instead of cloning
+        for contour in other.outline.contours() {
+            self.outline.push_contour(contour.clone());
+        }
+    }
+    
+    /// Transform the path with the given transformation.
+    /// returns the transformed path.
+    ///
+    /// note: any non-closed subpaths will be closed
+    pub fn transform(mut self, transform: &Transform2F) -> Path2D {
+        self.flush_current_contour();
+        self.outline.transform(transform);
+        self
+    }
```

Comment:
These two can probably be added without problems. They are used by [vector](https://github.com/s3bk/vector).